pkg/performanceprofile/controller/performanceprofile/status/writer.go

```diff
@@ -0,0 +1,126 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	mcov1 "github.com/openshift/api/machineconfiguration/v1"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ Writer = &writer{}
+
+type writer struct {
+	client.Client
+}
+
+func NewWriter(c client.Client) Writer {
+	return &writer{Client: c}
+}
+
+func (w *writer) Update(ctx context.Context, object client.Object, conditions []conditionsv1.Condition) error {
+	profile, ok := object.(*performancev2.PerformanceProfile)
+	if !ok {
+		return fmt.Errorf("wrong type conversion; want=PerformanceProfile got=%T", object)
+	}
+	return w.update(ctx, profile, conditions)
+}
+
+func (w *writer) UpdateOwnedConditions(ctx context.Context, object client.Object, profileMCP *mcov1.MachineConfigPool) error {
```

Comment:
how hard it is (perhaps in a later PR) to fetch the correct MCP inside this function instead of requiring as argument? Reason I'm asking is to look for consistent API between hypershift and non-hypershift. Per previous comments, looks like this argument should always be `nil` in hypershift, which suggest the API is not generic enough.

Reply:
Done